src/flat_bvh.rs

```diff
@@ -260,130 +106,65 @@ impl BVHNode {
     }
 }
 
-impl BVH {
-    /// Flattens the [`BVH`] so that it can be traversed iteratively.
-    ///
-    /// [`BVH`]: struct.BVH.html
-    ///
-    /// # Examples
-    ///
-    /// ```
-    /// use bvh::aabb::{AABB, Bounded};
-    /// use bvh::bvh::BVH;
-    /// use bvh::nalgebra::{Point3, Vector3};
-    /// use bvh::ray::Ray;
-    ///
-    /// # struct Sphere {
-    /// #     position: Point3<f32>,
-    /// #     radius: f32,
-    /// # }
-    /// #
-    /// # impl Bounded for Sphere {
-    /// #     fn aabb(&self) -> AABB {
-    /// #         let half_size = Vector3::new(self.radius, self.radius, self.radius);
-    /// #         let min = self.position - half_size;
-    /// #         let max = self.position + half_size;
-    /// #         AABB::with_bounds(min, max)
-    /// #     }
-    /// # }
-    /// #
-    /// # fn create_bounded_shapes() -> Vec<Sphere> {
-    /// #     let mut spheres = Vec::new();
-    /// #     for i in 0..1000u32 {
-    /// #         let position = Point3::new(i as f32, i as f32, i as f32);
-    /// #         let radius = (i % 10) as f32 + 1.0;
-    /// #         spheres.push(Sphere {
-    /// #             position: position,
-    /// #             radius: radius,
-    /// #         });
-    /// #     }
-    /// #     spheres
-    /// # }
-    /// #
-    /// let origin = Point3::new(0.0,0.0,0.0);
-    /// let direction = Vector3::new(1.0,0.0,0.0);
-    /// let ray = Ray::new(origin, direction);
-    /// let shapes = create_bounded_shapes();
-    /// let bvh = BVH::build(&shapes);
-    /// let flat_bvh = bvh.flatten();
-    /// ```
-    pub fn flatten(&self) -> Vec<FlatNode> {
-        let mut vec = Vec::new();
-        self.root.flatten(&mut vec, 0);
-        vec
-    }
+/// A flat [`BVH`]. Represented by a vector of [`FlatNode`]s. The [`FlatBVH`] is designed for use
+/// where a recursive traversal of a data structure is not possible.
```

Comment:
Add an example as to why this might be the case. IE "For instance on some architectures or devices (like GPUs) ..."